src/resmom/mom_pmix.c

```diff
@@ -0,0 +1,1537 @@
+ * Software Foundation, either version 3 of the License, or (at your option) any
+ * later version.
+ *
+ * PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Commercial License Information:
+ *
+ * For a copy of the commercial license terms and conditions,
+ * go to: (http://www.pbspro.com/UserArea/agreement.html)
+ * or contact the Altair Legal Department.
+ *
+ * Altair’s dual-license business model allows companies, individuals, and
+ * organizations to create proprietary derivative works of PBS Pro and
+ * distribute them - whether embedded or bundled with other software -
+ * under a commercial license agreement.
+ *
+ * Use of Altair’s trademarks, including but not limited to "PBS™",
+ * "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+ * trademark licensing policies.
+ *
+ */
+/**
+ * @file	mom_pmix.c
+ */
+
+#include <pbs_config.h>
+
+#ifdef PMIX
+
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <pthread.h>
+#include "mom_pmix.h"
+#include "mom_func.h"
+#include "list_link.h"
+#include "log.h"
+#include "tm.h"
+
+extern char *log_file;
+extern char *path_log;
+extern char mom_short_name[];
+extern pbs_list_head svr_alljobs;
+
+#if 0
+
+/*
+ * Some operations like spawn and fence are not atomic and occur
+ * over a series of steps. In some cases, data needs to be retained
+ * and used in subsequent steps. We might choose to define a data
+ * structure that houses both tracking information (e.g. namespace,
+ * operation type, etc.) together with an opaque union to house the
+ * underlying data. That is what is being suggested (but not yet
+ * implemented) here.
+ */
+
+/* Enumerate operations that require PBS to call back */
+typedef enum pbs_pmix_oper_type {
+	PBS_PMIX_OPER_NONE,
+	PBS_PMIX_FENCE,
+	PBS_PMIX_SPAWN,
+	/* Add new operations before PBS_PMIX_OPER_UNDEFINED */
+	PBS_PMIX_OPER_UNDEFINED
+} pbs_pmix_oper_type_t;
+
+/* Data structure to house auxiliary PMIx operation data */
+typedef struct pbs_pmix_oper {
+	pbs_pmix_oper_type_t type;
+	job *pjob;
+	struct pbs_pmix_oper *next;
+	union {
+		pbs_pmix_fence_data_t fence;
+		pbs_pmix_spawn_data_t spawn;
+	} data;
+} pbs_pmix_oper_t;
+
+#endif
+
+/* Locking structure and macros */
+typedef struct {
+	pthread_mutex_t mutex;
+	pthread_cond_t cond;
+	volatile bool active;
+	pmix_status_t status;
+} pbs_pmix_lock_t;
+
+#define PBS_PMIX_CONSTRUCT_LOCK(l) \
+	do { \
+		pthread_mutex_init(&(l)->mutex, NULL); \
```

Comment:
I haven't answered your question yet because I'm not convinced I have a good answer. There's not much we can do if the call fails. These macros were patterned after some PMIx code that also ignores the return value. I think the best answer I can provide at the moment is that it hasn't been a problem for PMIx so far, and we can enhance the code in the future if need be.